### PR TITLE
Add support for different drives in Windows

### PR DESCRIPTION
**Pull request by pablonete**
So if code is in D: but temporary folder is in C:, don't try to find a relative path but stay on the absolute one.
@arcanis Please review and merge, this should fix test `it should correctly create resolution mounting points when using the link protocol`, I hope it doesn't break others.

### CODE REVIEW

**Comment by pablonete** [comment]
@arcanis I believe we don't need to change normalizePath, it detects absolute correctly with any slashes in Windows:
```
> path.isAbsolute("x:\\hola")
true
> path.isAbsolute("x:/hola")
true
```
This is using `path.win32`, while using `path.posix` none of them would be true.

I tested your proposal anyway, found a problem with using `name` as it gets rid of the file extension (so `no-deps-1.0.0` became `no-deps-1.0`). Using `base` instead of `name`, it works, but I still believe we don't need the extra complexity. Let me know if I'm missing something.

File: src/util/portable-script.js [hunk 1]
```diff
@@ -36,11 +36,20 @@ async function makePortableProxyScriptUnix(
...
-  await fs.writeFile(
-    filePath,
-    `#!/bin/sh\n\n${environment}"${sourcePath}"${prependedArguments} "$@"${appendedArguments}\n`,
-  );
-  await fs.chmod(filePath, 0o755);
+
+  if (process.platform === 'win32') {
+    await fs.writeFile(
+      filePath + '.cmd',
+      `@${environment}"${sourcePath}" ${prependedArguments} ${appendedArguments} %*\r\n`,
```
Comment (pablonete) [inline src/util/portable-script.js]: I've merged approaches from linux branch of this `if` with what cmd-shim does (that's where `%*` is coming from). But I'd need @arcanis to double check I'm not doing anything wrong here. Thanks.

**Comment by arcanis** [comment]
Thanks Pablo! I'm at dotjs today, but I'll take a look asap

**Comment by pablonete** [comment]
Locally I've got a 100% with my last escape fix (which I wonder why it didn't fail before):

```
Test Suites: 1 passed, 1 total
Tests:       1 skipped, 132 passed, 133 total
Snapshots:   0 total
Time:        405.318s
Ran all test suites.
Done in 406.63s.
```

Looking forward to seeing results in Linux and Azure Pipelines once this is reviewed. Enjoy DotJS!

**Pull request merged**